documentation/bolt_examples.md

```diff
@@ -123,3 +123,216 @@ To convert the `restart_service.ps1` script to a task, giving you the ability to
+
+1 . Create a file named `Puppetfile` in the project directory, with the modules needed for this example:
+
+```     
+# Modules from the Puppet Forge
+mod 'puppetlabs-chocolatey', '4.1.0'
+## dependencies
+mod 'puppetlabs-stdlib', '4.13.1' #install latest
+mod 'puppetlabs-powershell', '2.3.0'
+mod 'puppetlabs-registry', '2.1.0'
+        
+# Modules from Git
+# mod 'apache',
+# :git    => 'https://github.com/puppetlabs/puppetlabs-apache',
+# :commit => 'de290646f97e04b4b8e42c70f6e01e860c394ce7'     
+ ```
+
+Note that you can install modules from a number of different sources. For more information, see the [Puppetfile readme](https://github.com/puppetlabs/r10k/blob/master/doc/puppetfile.mkd#examples).
+
+2. Add the Puppetfile to a remote repository, for example, GitHub.
+
+3. From inside the project directory, install the required modules:
+
+```
+bolt puppetfile install
+```
+
+After it runs, you can see a `modules` directory inside the project directory, containing the modules you specified in the `Puppetfile`.
+
+### 3. Write a Bolt plan to apply Puppet code and orchestrate the deployment of a package resource using the Chocolatey provider. Plans allow you to run more than one task with a single command.
+
+1. Create a `site-modules` directory. This is where you will add code and local modules.
+
+2. Inside the `site-modules` directory, create a new module called `puppet_choco_tap`.
+
+3. Inside the `puppet_choco_tap` module, create a plans directory with an `init.pp file` called `/plans/installer.pp`.
+
+The folder tree should look like this:
+
+```
+├──plans
+   └── installer.pp 
+```
+
+2. Create a plan called `puppet_chocolatey_tap::installer` by copying the following code into the `installer.pp` file::
+
+```
+plan puppet_choco_tap::installer(
+  TargetSpec $nodes,
+  String $package,
+  Variant[Enum['absent', 'present'], String ] $ensure = 'present',
+){
+  apply_prep($nodes)
+
+  apply($nodes){
+    include chocolatey
+
+    package { $package :
+      ensure    => $ensure,
+      provider  => 'chocolatey',
+      }
+    }
+  }
+}
+```
+Take note of the following features of the plan:
+
+- It has three parameters: the `TargetSpec`, a `package` string for the package name, and the `ensure` state of the package which allows for version, absent or present.
+- It has the `apply_prep` function call, which is used to install modules needed by `apply` on remote nodes as well as to gather facts about the nodes.
+- The first `apply` block installs the Chocolatey package manager. The Chocolatey provider is also deployed as a library with the Puppet agent in `apply_prep`.
+- The second `apply` block installs a specified package using the Chocolatey provider.
+
+2. To verify that the `puppet_choco_tap::installer` plan is available, add and commit your code to your remote repository, and run the following command again:
```

Comment:
You will not need to commit code and re-run puppetfile install. 
```suggestion
2. To verify that the `puppet_choco_tap::installer` plan is available:
```
then remove `bolt puppetfile install` and keep `bolt plan show`.

Reply:
@clairecadman Same here - Please add Cas' update